### PR TITLE
win_nssm: Clean up parameter handling + docs

### PR DESCRIPTION
**Pull request by dagwieers**
##### ISSUE TYPE

 - Bugfix Pull Request

##### COMPONENT NAME

win_nssm

##### ANSIBLE VERSION

v2.3

##### SUMMARY

Changes include:
- Use Get-AnsibleParam with -type
- Replace $result PSObject with normal hash
- Clean up documentation

### CODE REVIEW

File: lib/ansible/modules/windows/win_nssm.ps1 [hunk 1]
```diff
@@ -23,25 +23,26 @@ $ErrorActionPreference = "Stop"
...
-$result = New-Object PSObject;
-Set-Attr $result "changed" $false;
+$result = @{
+    changed = $false
+}
 
-$name = Get-Attr $params "name" -failifempty $true
-$state = Get-Attr $params "state" -default "present" -validateSet "present", "absent", "started", "stopped", "restarted" -resultobj $result
+$name = Get-AnsibleParam -obj $params -name "name" -type "str" -failifempty $true
+$state = Get-AnsibleParam -obj $params -name "state" -type "str" -default "present" -validateset "present","absent","started","stopped","restarted" -resultobj $result
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_nssm.ps1]: Is the -resultobj required?

File: lib/ansible/modules/windows/win_nssm.ps1 [hunk 2]
```diff
@@ -23,25 +23,26 @@ $ErrorActionPreference = "Stop"
...
 
-$application = Get-Attr $params "application" -default $null
-$appParameters = Get-Attr $params "app_parameters" -default $null
-$appParametersFree  = Get-Attr $params "app_parameters_free_form" -default $null 
-$startMode = Get-Attr $params "start_mode" -default "auto" -validateSet "auto", "manual", "disabled" -resultobj $result
+$application = Get-AnsibleParam -obj $params -name "application" -type "str"
+$appParameters = Get-AnsibleParam -obj $params -name "app_parameters" -type "str"
+$appParametersFree  = Get-AnsibleParam -obj $params -name "app_parameters_free_form" -type "str"
+$startMode = Get-AnsibleParam -obj $params -name "start_mode" -type "str" -default "auto" -validateset "auto","manual","disabled" -resultobj $result
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_nssm.ps1]: -resultobj again?

File: lib/ansible/modules/windows/win_nssm.ps1 (hunk 1 above)
Comment (jborean93) [inline lib/ansible/modules/windows/win_nssm.ps1:33]: Is the -resultobj required?

File: lib/ansible/modules/windows/win_nssm.ps1 (hunk 2 above)
Comment (jborean93) [inline lib/ansible/modules/windows/win_nssm.ps1:38]: -resultobj again?

File: lib/ansible/modules/windows/win_nssm.ps1 [hunk 3]
```diff
@@ -544,16 +545,16 @@ Function Nssm-Start
...
+        $result.nssm_error_log = "$results"
         Throw "Error starting service ""$name"""
     }
 
     switch ($currentStatus)
     {
         "SERVICE_RUNNING" { <# Nothing to do #> }
         "SERVICE_STOPPED" { Nssm-Start-Service-Command -name $name }
-        
+
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_nssm.ps1:557]: Random newline?

**Review by jborean93 — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/windows/win_nssm.ps1 (hunk 3 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_nssm.ps1:557]: Nope, just a cleanup of trailing whitespaces. My editor makes them quite obvious, so I remove them.

File: lib/ansible/modules/windows/win_nssm.ps1 (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_nssm.ps1:33]: Yup, prefer to not touch those until we have decided what to do with them.

File: lib/ansible/modules/windows/win_nssm.ps1 (hunk 2 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_nssm.ps1:38]: Yup, prefer to not touch those until we have decided what to do with them.

**Comment by dagwieers** [comment]
!needs_revision

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
shipit

**Review by nitzmahone — APPROVED** [review]

**Pull request merged**